File: src/main/java/org/faystmax/compliment/bot/bot/ComplimentTelegramBot.java
package org.faystmax.compliment.bot.bot;

import lombok.SneakyThrows;
import lombok.extern.slf4j.Slf4j;
import org.telegram.telegrambots.bots.DefaultBotOptions;
import org.telegram.telegrambots.bots.TelegramLongPollingBot;
import org.telegram.telegrambots.meta.api.methods.send.SendMessage;
import org.telegram.telegrambots.meta.api.objects.Update;
import org.telegram.telegrambots.meta.exceptions.TelegramApiException;

import java.util.List;
import java.util.concurrent.ThreadLocalRandom;

@Slf4j
public class ComplimentTelegramBot extends TelegramLongPollingBot {
    private final String username;
    private final List<String> compliments;

    public ComplimentTelegramBot(final String username, final String botToken, final List<String> compliments, final DefaultBotOptions options) {
        super(options, botToken);
        this.username = username;
        this.compliments = compliments;
    }

    @Override
    public void onUpdateReceived(final Update update) {
        try {
            log.info("UpdateReceived chatId = {}", update.getMessage().getChatId());

            final String compliment = compliments.get(ThreadLocalRandom.current().nextInt(0, compliments.size()));
            sendApiMethod(new SendMessage(update.getMessage().getChatId().toString(), compliment));
            log.info("Sending compliment = '{}' to chatId = '{}'", compliment, update.getMessage().getChatId());
        } catch (final TelegramApiException e) {
            throw new RuntimeException(e);
        }
    }

    @Override
    public String getBotUsername() {
        return username;
    }

    @Override
    public void onRegister() {
        log.info(username + " registered successfully!");
    }

    @SneakyThrows
    public void sendCompliment(final String chatId) {
        final String compliment = compliments.get(ThreadLocalRandom.current().nextInt(0, compliments.size()));
        log.info("Sending compliment to chatId = '{}'. compliment = '{}'", chatId, compliment);
        sendApiMethod(new SendMessage(chatId, compliment));
    }
}

File: src/main/kotlin/bot/ComplimentTelegramBot.kt
package org.faystmax.bot

import org.telegram.telegrambots.bots.DefaultBotOptions
import org.telegram.telegrambots.bots.TelegramLongPollingBot
import org.telegram.telegrambots.meta.api.methods.send.SendMessage
import org.telegram.telegrambots.meta.api.objects.Update
import kotlin.random.Random

class ComplimentTelegramBot(
    val username: String,
    botToken: String,
    val compliments: List<String>,
    options: DefaultBotOptions,
) : TelegramLongPollingBot(options, botToken) {

    override fun getBotUsername(): String = username

    override fun onRegister() {
        println("$username registered successfully!")
    }

    override fun onUpdateReceived(update: Update) {
        sendApiMethod(SendMessage(update.message.chatId.toString(), compliments.get(Random.nextInt(0, 60))))
        println(update.message.chatId)
    }

    fun sendMsg(chatId: String, msg: String) {
        sendApiMethod(SendMessage(chatId, msg))
    }
}